### DISCUSSION TITLE
How to specify begin and end option?

### DISCUSSION THREAD

**Discussion by Thesharing** [async-profiler/async-profiler#430] [Q&A] [answered]
I'm trying to use async-profiler to capture samples starting from a certain function. I specified async-profiler in JVM options like this: 

```bash 
-agentpath:/tmp/async-profiler/build/libasyncProfiler.so=file=<LOG_DIR>/flame.svg,begin=akka/remote/EndpointWriter.writeSend
```

But it doesn't work. There's nothing in the output svg file. Is the format of methods in `begin` option wrong?

**Comment by apangin** [comment, marked as answer]
From the documentation:

> --begin function, --end function - automatically start/stop profiling when the specified *native* function is executed.

`EndpointWriter.writeSend` is a Scala method, not a *native* function.

The use case for `--begin/--end` is different - it's main purpose is to profile certain JVM phases like GC and Safepoint pauses. If your goal was to profile a particular application method, perhaps you wanted to use filters (`include`, `exclude`) instead.

Also note that `.svg` output is no longer supported. Starting from async-profiler 2.0, Flame Graphs are in `.html` format.

**Reply by Thesharing** [reply]
Thank you for the detailed explanation🙏. Is `--begin/--end` only supported in 2.0?

**Reply by apangin** [reply]
Yes
